sertop/sercomp.ml

```diff
@@ -124,6 +124,14 @@ let process_vernac ~mode ~pp ~doc ~sid ast =
     | C_sexp ->
       printf "@[%a@]@\n%!" pp
         Serlib.(Ser_cAst.sexp_of_t Ser_vernacexpr.sexp_of_vernac_control ast)
+    | C_goals ->
+       ignore (Stm.observe ~doc:doc n_st);
+       let sg_pre = Serapi_goals.get_goals ~doc:doc n_st in
+       match sg_pre with
+       | Some g ->
+	  let Serapi_goals.{ goals; stack; _ } = g in
+	  printf "%d %d\n%!" (List.length goals) (List.length stack)
```

Comment:
Some funny indentation here? Also the `doc` returned by `observe` should not be ignored but used in subsequent calls.